01_Libraries/ghost_v5_interfaces/include/ghost_v5_interfaces/devices/digital_device_interface.hpp

```diff
@@ -0,0 +1,129 @@
+/*
+ *   Copyright (c) 2024 Maxx Wilson, Xander Wilson
+ *   All rights reserved.
+
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+
+ *   The above copyright notice and this permission notice shall be included in all
+ *   copies or substantial portions of the Software.
+
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *   SOFTWARE.
+ */
+
+#pragma once
+
+#include <cstring>
+#include "ghost_v5_interfaces/devices/device_interfaces.hpp"
+
+namespace ghost_v5_interfaces
+{
+
+namespace devices
+{
+
+class DigitalDeviceData : public DeviceData
+{
+public:
+  struct SerialConfig
+  {
+    SerialConfig()
```

Comment:
If this only matters to distinguish input vs output, why not just have those as two separate device classes since you already have each port as its own device?

Reply:
Looks like this got resolved.